### PR TITLE
[IA-3983] Add leo.conf and support to run leo locally with new conf

### PR DESCRIPTION
**Pull request by jdcanas**
This PR accomplishes the following:
- Adds a local leo.conf with HOCON variable substitution that allows ENV vars to populate the conf
- Modifies Leo's `Config.scala` to load this new config as the primary, while maintaining backwards compatibility with firecloud-develop config
- Adds support for running leo locally with this new config (`local.env`), effectively removing the `docker-rsync-local` script from firecloud-develop. This includes secret file generation for values injected into the conf from vault
- Separately updates the documentation in confluence on how to run leo locally (already done)

There will be a follow-up PR in firecloud-develop to remove `docker-rsync-local-leonardo.sh`, and one in `terra-helmfile` to start populating this env vars

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/Config.scala [hunk 1]
```diff
@@ -54,7 +54,27 @@ import scala.concurrent.duration._
...
+  /** Loads all the configs for the Leo App. All values defined in `src/main/resources/leo.conf` will take precedence over any other configs. In this way, we
+   * can still use configs rendered by `firecloud-develop` that render to `config/leonardo.conf` if we want.
+   *     If you want to use leo.conf you must populate the appropriate ENV vars (done in terra-helmfile)
+   *     leonardo.conf is what firecloud-develop renders, and it will be used if the ENV vars are not populated for leo.conf
+   */
+  // Config that lives in leo repo in /src/main/resources
+  val leoConfig = ConfigFactory.parseResourcesAnySyntax("leo")
+
+  // Config that is generated to /config from firecloud-develop
+  val backupConfig1 = ConfigFactory
```
Comment (jyang-broad) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/Config.scala]: nit. does this make sense to call firecloudDevelopConfig?

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/Config.scala [hunk 2]
```diff
@@ -54,7 +54,27 @@ import scala.concurrent.duration._
...
+    .parseResources("leonardo.conf")
+
+  // Load any other configs on the classpath following: https://github.com/lightbend/config#standard-behavior
+  // This is where things like `src/main/resources/reference.conf` will get loaded
+  val backupConfig2 = ConfigFactory.load()
+
+  val config = leoConfig
```
Comment (jyang-broad) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/Config.scala:75]: would love a comment here on which config has priority to overwrite at a glance.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/Config.scala [hunk 3]
```diff
@@ -54,7 +54,27 @@ import scala.concurrent.duration._
...
+  // Config that lives in leo repo in /src/main/resources
+  val leoConfig = ConfigFactory.parseResourcesAnySyntax("leo")
+
+  // Config that is generated to /config from firecloud-develop
+  val backupConfig1 = ConfigFactory
+    .parseResources("leonardo.conf")
+
+  // Load any other configs on the classpath following: https://github.com/lightbend/config#standard-behavior
+  // This is where things like `src/main/resources/reference.conf` will get loaded
+  val backupConfig2 = ConfigFactory.load()
```
Comment (jyang-broad) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/Config.scala]: nit. would like a more meaningful name. referenceConfig? 🤷

File: env/local.env [hunk 4]
```diff
@@ -0,0 +1,99 @@
+# Note these SHOULD NOT be quoted, as all values in this file are interpreted as literals directly as they appear when used in conjunction with dockers `--env-file` option
```
Comment (jyang-broad) [inline env/local.env:1]: nice comment.

File: docker-rsync-local-leonardo.sh [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    docker cp config/jupyter-server.crt leonardo-sbt:/etc/jupyter-server.crt
+    docker cp config/jupyter-server.key leonardo-sbt:/etc/jupyter-server.key
+    docker cp config/leo-client.p12 leonardo-sbt:/etc/leo-client.p12
+    docker cp config/rootCA.key leonardo-sbt:/etc/rootCA.key
+    docker cp config/rootCA.pem leonardo-sbt:/etc/rootCA.pem
+    docker cp /etc/localtime leonardo-sbt:/etc/localtime
+    docker cp leonardo-helm-lib:/build /tmp
+    docker cp /tmp/build/ leonardo-sbt:/helm-go-lib-build
+#    docker cp env/local.env leonardo-sbt:/etc/local.env
+#    dokcer cp rendered/secrets.env leonardo-sbt:/etc/secrets.env
```
Comment (jyang-broad) [inline docker-rsync-local-leonardo.sh]: typo. also, are these working/testing things that should be removed before commit or switches for special dev customization

File: docker-rsync-local-leonardo.sh [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+}
+echo "Performing initial file sync..."
+run_rsync
+fswatch -o . | while read f; do run_rsync; done &
+echo $$ > ./.docker-rsync-local.pid
+
+start_server () {
+    docker network create fc-leonardo
+
+    echo "Creating Google sqlproxy container..."
```
Comment (jyang-broad) [inline docker-rsync-local-leonardo.sh:65]: just curious, where does this db live (assuming this is connecting to cloudsql somewhere)? Is it shared or does everyone have their own.

File: http/src/main/resources/leo.conf [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+  # See reference.conf for default liquibase.changelog value
+  initWithLiquibase = ${?SHOULD_INIT_WITH_LIQUIBASE}
+}
+
+sam {
+  server = ${?SAM_URL}
+}
+
+akka {
+  loglevel = INFO
```
Comment (jyang-broad) [inline http/src/main/resources/leo.conf]: can we set this to a env var for debugging?

**Review by jyang-broad — APPROVED** [review]
Added some comments/asks on clarity but looks good!

File: docker-rsync-local-leonardo.sh (hunk 6 above)
Comment (jdcanas) [reply docker-rsync-local-leonardo.sh:65]: Yeah, theres a `sqlproxy.env` file generated by firecloud-develop. By default, local firecloud-develop generates this file pointing to the dev DB. However, the instructions for how to run leo locally specify that devs should clone the dev DB and point their `sqlproxy.env` file there

**Comment by jdcanas** [comment]
jenkins retest

**Pull request merged**